inhouse_bot/cogs/admin_cog.py

```diff
@@ -167,3 +170,48 @@ async def config(self, ctx: commands.Context, config_key: str, option: str = "")
 
             value = 'ON' if server_config.config.get(config_key) else 'OFF'
             await ctx.send(f"{config_key} is: {value}")
+
+    @admin.command()
+    @guild_only()
+    @doc(f"""
+        Connects a Summoner to a discord account
+
+        Example:
+            `{PREFIX}admin verify @User <Summoner Name>`
+    """)
+    async def verify(self, ctx: commands.Context, user: discord.Member, name: str = ""):
+        # did not provide discord member
+        if not user:
+            return await ctx.send("Missing user.")
+
+        # did not provide summoner name
+        if not name:
+            return await ctx.send("Missing Summoner name.")
+
+        id = ctx.author.id
+        server_id = ctx.guild.id
+
+        summoner = self.lol_Api.get_summoner_by_name(name)
+        puuid = summoner["puuid"]
+        summonerName = summoner["name"]
+
+        with session_scope() as session:
+            player = (
+                session.query(Player)
+                .select_from(Player)
+                .filter(Player.server_id == server_id)
+                .filter(Player.summoner_puuid == puuid)
+            ).one_or_none()
+
+            if player:
+                if player.id == id:
+                    return await ctx.send("Summoner is already verified")
+                else:
+                    (session.query(Player)
+                    .filter(Player.id == player.id)
+                    .filter(Player.server_id == server_id)
+                    .update({ Player.summoner_puuid: None }))
+
+            session.merge(Player(id=user.id, server_id=server_id, name=name, summoner_puuid=puuid))
+            
+        await ctx.send(f"Verified Summoner name: {summonerName}")
```

Comment:
So 2 things 
1: I think the code could be reduced to what I posted in discord, whats the intent behind the else block?

2: Also I think a slight bug or maybe its with intent, but when you insert into player table it does it based on their summoners rift name, is the intent this so for the tournament draft code?

Reply:
The else occurs when there is another player that already has that puuid set. It is necessary to update that player entity and set the puuid to NULL otherwise there'd be two different players on the same server with the same puuid and break the table constraint.

for 2: yes, it searches summoner by name. I think this is the best way to get the summoner